Purchases/Public/Purchases.swift

```diff
@@ -802,16 +802,16 @@ public extension Purchases {
      * If the fetch fails for any reason it will return an empty array.
      */
     @objc(getProductsWithIdentifiers:completion:)
-    func getProducts(_ productIdentifiers: [String], completion: @escaping ([SKProduct]) -> Void) {
+    func getProducts(_ productIdentifiers: [String], completion: @escaping ([StoreProduct]) -> Void) {
```

Comment:
We can't have an overload with the same name, we'd have to have a different name (Swift allows overloads with different types but Objective-C doesn't).
Right now users would get a compile error saying "wrong type", which I think it might be almost a useful as a fix it to point to a different method, that will also have the different type?

Reply:
🤔 thinking about this more, this may even work for most folks without any changes: if you don't explicitly specify the type you're expecting in the completion block, and you're using properties that are also present in `SKProduct`, it'd just work. 

like, 

```swift
    let products = try! await getProducts(["a"])
```
will still work unless you either explicitly specify the type or try to use a property that isn't present in `StoreProduct` (same for completion blocks syntax)

Reply:
Right exactly :)